### PR TITLE
Convert text/template Usage to fmt.Sprintf()

### LINKED ISSUE #7456: Tech Debt: Convert text/template Usage to fmt.Sprintf()
Currently there is inconsistent usage of `text/template` in a few of the acceptance testing files to build test Terraform configurations, e.g.

https://github.com/terraform-providers/terraform-provider-aws/blob/20465b66820363029e31428e53a1b1714c932835/aws/resource_aws_budgets_budget_test.go#L314-L425

https://github.com/terraform-providers/terraform-provider-aws/blob/20465b66820363029e31428e53a1b1714c932835/aws/resource_aws_s3_bucket_test.go#L2021-L2087

They will soon be fixed to at least perform error checking, but for consistency with the rest of the codebase and to simplify the handling of building test configurations, these should be converted to `fmt.Sprintf()` or similar.

### PR DESCRIPTION
**Pull request by kterada0509**
Fixes #7456 

Changes proposed in this pull request:

* Convert text/template Usage to fmt.Sprintf in `aws_s3_bucket` resource test
* Convert text/template Usage to fmt.Sprintf in `aws_budgets_budget` resource test

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSS3MultiBucket_withTags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSS3MultiBucket_withTags -timeout 120m
=== RUN   TestAccAWSS3MultiBucket_withTags
=== PAUSE TestAccAWSS3MultiBucket_withTags
=== CONT  TestAccAWSS3MultiBucket_withTags
--- PASS: TestAccAWSS3MultiBucket_withTags (44.62s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	44.793s
```

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSBudgetsBudget_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSBudgetsBudget_ -timeout 120m
=== RUN   TestAccAWSBudgetsBudget_basic
=== PAUSE TestAccAWSBudgetsBudget_basic
=== RUN   TestAccAWSBudgetsBudget_prefix
=== PAUSE TestAccAWSBudgetsBudget_prefix
=== CONT  TestAccAWSBudgetsBudget_basic
=== CONT  TestAccAWSBudgetsBudget_prefix
--- PASS: TestAccAWSBudgetsBudget_prefix (35.01s)
--- PASS: TestAccAWSBudgetsBudget_basic (38.60s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws    38.670s
```

### CODE REVIEW

File: aws/resource_aws_s3_bucket_test.go [hunk 1]
```diff
@@ -2021,74 +2019,68 @@ resource "aws_s3_bucket" "bucket" {
...
 }
-`))
-	var doc bytes.Buffer
-	// TODO: Convert to fmt.Sprintf() (https://github.com/terraform-providers/terraform-provider-aws/issues/7456)
-	if err := t.Execute(&doc, struct{ GUID int }{GUID: randInt}); err != nil {
-		panic(fmt.Sprintf("error executing template: %s", err))
-	}
-	return doc.String()
+
+`, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt, randInt)
```
Comment (bflad) [inline aws/resource_aws_s3_bucket_test.go]: Drive-by comment: You can use indexes in `fmt` to prevent the sprawling argument problem. 😄 

```go
fmt.Sprintf(`
foo = %[1]d
bar = %[1]d
`, 1234)
```

Comment (kterada0509) [reply aws/resource_aws_s3_bucket_test.go]: @bflad 
Thanks I Fixed it.

**Comment by kterada0509** [comment]
Rerun test.

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSS3MultiBucket_withTags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSS3MultiBucket_withTags -timeout 120m
=== RUN   TestAccAWSS3MultiBucket_withTags
=== PAUSE TestAccAWSS3MultiBucket_withTags
=== CONT  TestAccAWSS3MultiBucket_withTags
--- PASS: TestAccAWSS3MultiBucket_withTags (45.71s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	45.802s
```
